tool2schema/parameter_schema.py

```diff
@@ -145,21 +187,94 @@ def matches(parameter: Parameter) -> bool:
         args = typing.get_args(parameter.annotation)
         return (
             parameter.annotation != parameter.empty
-            and typing.get_origin(parameter.annotation) is typing.Union
+            and typing.get_origin(parameter.annotation) is Union
             and len(args) == 2
             and type(None) in args
         )
 
-    def _add_type(self, schema: dict):
-        if sub_type := super()._get_sub_type():
-            schema["type"] = sub_type
+    def _get_type(self) -> Union[str, Parameter.empty]:
+        return super()._get_sub_type()
+
+
+class EnumParameterSchema(ParameterSchema):
+    """
+    Parameter schema for enumeration types.
+    """
+
+    def __init__(self, values: list, parameter: Parameter, index: int, docstring: str = None):
+        super().__init__(parameter, index, docstring)
+        self.enum_values = values
+
+    def _get_type(self) -> Union[str, Parameter.empty]:
+        return TYPE_MAP.get(type(self.enum_values[0]).__name__, "object")
+
+    def _get_enum(self) -> Union[list[str], Parameter.empty]:
+        return self.enum_values
+
+
+class EnumTypeParameterSchema(EnumParameterSchema):
+    """
+    Parameter schema for enum.Enum types.
+    """
+
+    def __init__(self, parameter: Parameter, index: int, docstring: str = None):
+        values = [e.name for e in parameter.annotation]
+        super().__init__(values, parameter, index, docstring)
+
+    @staticmethod
+    def matches(parameter: Parameter) -> bool:
+        return (
+            parameter.annotation != parameter.empty
+            and isclass(parameter.annotation)
+            and issubclass(parameter.annotation, Enum)
+        )
+
+    def encode_value(self, value):
+        """
+        Convert an enum instance to its name.
+
+        :param value: The enum instance to be converted.
+        """
+        return value.name
```

Comment:
Might be worth checking the type before calling `value.name`?

Reply:
Done in ad025df, replaced with 
```python
return value.name if isinstance(value, Enum) else value
```
so that the user can write `def myFunc(b: CustomEnum = "B")` and the method does not throw an exception